src/aws-sdk/types.ts

```diff
@@ -0,0 +1,103 @@
+import type { Service as AWSService } from "aws-sdk";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import type * as AWS from "aws-sdk";
+import { AnyFunction, OverloadUnion } from "../util";
+
+type AWSServiceClass = { new (): AWSService };
+
+/**
+ * First we have to extract the names of all Services in the v2 AWS namespace
+ *
+ * @returns "AccessAnalyzer" | "Account" | ... | "XRay"
+ */
+export type ServiceKeys = Exclude<
+  {
+    [K in keyof typeof AWS]: typeof AWS[K] extends AWSServiceClass ? K : never;
+  }[keyof typeof AWS],
+  "Service"
+>;
+
+export type SDK = {
+  [serviceName in ServiceKeys]: typeof AWS[serviceName] extends new (
+    ...args: any[]
+  ) => infer Client
+    ? {
+        [methodName in keyof Client]: SdkMethod<Client[methodName]>;
+      }
+    : never;
+};
```

Comment:
@thantos we should try collapsing this to an interface. This may help with redundant type computation

Reply:
```
type Mixin<T> = T

interface SDK extends Mixin<{
  [serviceName in keyof typeof AWS]: ...
}
```

Reply:
Hmm this might not help because it's methods that have the problem and they will still be lazily bound